tools/cbmc/proofs/HTTP/IotHttpsClient_ReadHeader/IotHttpsClient_ReadHeader_harness.c

```diff
@@ -11,18 +11,25 @@
 
 void harness() {
   IotHttpsResponseHandle_t respHandle = allocate_IotResponseHandle();
-  size_t nameLen;
-  size_t valueLen;
-  char *pName = safeMalloc(nameLen);
-  char  *pValue = safeMalloc(valueLen);
-  initialize_IotResponseHandle(respHandle);
   if (respHandle) {
     __CPROVER_assume(is_valid_IotResponseHandle(respHandle));
     // Until a CBMC issue is addressed
     // https://github.com/diffblue/cbmc/issues/5004
     // copy an assumption from is_valid above as a simple assignment below.
     respHandle->httpParserInfo.readHeaderParser.data = respHandle;
   }
-  __CPROVER_assume(0 < valueLen && valueLen < UINT32_MAX);
+  initialize_IotResponseHandle(respHandle);
+
+  size_t nameLen;
+  size_t valueLen;
+  __CPROVER_assume(nameLen < CBMC_MAX_OBJECT_SIZE);
+  __CPROVER_assume(valueLen < CBMC_MAX_OBJECT_SIZE);
+  char *pName = safeMalloc(nameLen+1);
```

Comment:
Why plus 1?  This is a mistake.  Not all strings are null terminated in this code.  But I see the original AddHeader above uses plus 1.

Reply:
I made the change to be consistent with AddHeader harness, but you are right, +1 is not necessary and the functions don't require the strings to be null-terminated. Looking at the AddHeader harness, I thought they do and might go out of bound when searching for the end, but they don't and CBMC is happy.

Reply:
I think I didn't test properly on my local branch. CBMC isn't happy.  It (within `strncmp`) requires the strings to be null terminated.

I have added the `+1` to both AddHeader and ReadHeader. Let me  know what you think.

Reply:
This requires review by the author @sarenameas:  Should the length be the string length of name and value or the actual byte length of the buffer containing name and value.

Reply:
the nameLen passed into ReadHeader() is the length of the name of the header field. Since pName is unbounded we have no idea how long the name in the pName buffer is going to be, so this look correct to me..

the valueLen passed into ReadHeader() is the size of the buffer pValue.

Reply:
@markrtuttle: I discussed this issue offline with @sarenameas and it turns out there is a bug in `IotHttpsClient_AddHeader` implementation in `iot_https_client.c`.

Thank you both for the thorough review!